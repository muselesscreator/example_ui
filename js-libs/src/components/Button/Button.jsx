/** @module */
import React from 'react';
import { Button, Icon } from 'semantic-ui-react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import Component from '../Component';

import './Button.scss';

/**
 * Basic Button implementation.  Adds our styling options to the basic
 * semantic-ui-react Button.
 *
 * Accepts all options for the semantic-ui-button:
 * <https://react.semantic-ui.com/elements/button>
 *
 * @param {string} [className] - additional css class names
 * @param {string} [icon] - icon name
 * @param {boolean} [dark=false] - is the button active? (adds css class)
 * @param {boolean} [active=false] - is the button active? (adds css class)
 * @param {boolean} [danger=false] - is the button active? (adds css class)
 * @param {boolean} [secondary=false] - is the button active? (adds css class)
 */
export class RCButton extends Component {
  render() {
    const {
      className,
      children,
      active,
      danger,
      dark,
      icon,
      secondary,
      ...props
    } = this.props;
    const optClassnames = { active, dark, danger, secondary };
    const btnProps = {
      size: 'mini',
      className: classNames(className, 'brw-btn', optClassnames),
      ...props,
    };
    if (children === undefined) {
      return <Button icon={icon} { ...btnProps } />
    }
    return (
      <Button { ...btnProps } >
        { icon !== undefined && <Icon name={icon} /> }
        { children }
      </Button>
    );
  }
}
RCButton.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.string,
  dark: PropTypes.bool,
  active: PropTypes.bool,
  danger: PropTypes.bool,
  secondary: PropTypes.bool,
}

RCButton.defaultProps = {
  active: false,
  className: '',
  danger: false,
  dark: false,
  secondary: false,
}

export default RCButton;
