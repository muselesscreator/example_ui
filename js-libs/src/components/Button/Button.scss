@import "../../scss/variables.scss";
@import "../../scss/colors.scss";

.ui.button.brw-btn {
  position: relative;
  height: 40px;
  -webkit-box-sizing: border-box;

  font-size: 14px !important;
  font-weight: normal !important;
  text-shadow: none !important;

  color: $dark_grey_50;
  background-color: $primary_30; 
  border: none !important;
  border-radius: 0px !important;

  &::before {
    height: 42px;
  }
  &:hover {
    color: $dark_grey_70;
    background-color: $primary_40;
    &.dark {
      background-color: $primary_40;
    }
  }
  &.active {
    color: $dark_grey_70;
    background-color: $primary_40;
    &.dark {
      background-color: $primary_40;
    }
  }

  &:focus::after {
      content: "";
      outline: 1px solid $primary_20;
      height: 100%;
      width: 100%;
      position: absolute;
      left: 0px;
      top: 0px;
  }
  &.disabled {
    background-color: $light_grey_50 !important;
    color: $light_grey_80 !important;
    opacity: 1 !important;
  }
  &.secondary {
    &:focus::after {
        height: calc(100% + 4px);
        width: calc(100% + 4px);
        left: -2px;
        top: -2px;
    }
    background-color: transparent;
    color: $dark_grey_50;
    border: solid 1px $light_grey_50 !important;
    height: 38px;
    &:hover {
      background-color: $light_grey_30;
    }
    &.active {
      background-color: $light_grey_40;
      color: $dark_grey_70;
    }
  }
  &.danger {
    color: $light_grey_20;
    background-color: $red_40;
    &:hover {
      color: $red_50;
    }
  }
}
