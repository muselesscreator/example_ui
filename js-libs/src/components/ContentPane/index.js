import ContentPane from './ContentPane';

export default ContentPane;
