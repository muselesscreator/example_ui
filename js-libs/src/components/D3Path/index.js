import D3Path from './D3Path';

export default D3Path;
