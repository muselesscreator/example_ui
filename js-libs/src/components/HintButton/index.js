import HintButton from './HintButton';

export default HintButton;
