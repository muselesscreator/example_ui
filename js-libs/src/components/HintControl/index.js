import HintControl from './HintControl';

export default HintControl;
