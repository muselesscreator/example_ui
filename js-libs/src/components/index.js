export { default as Accordion } from './Accordion';
export { default as Button } from './Button';
export { default as Canvas } from './Canvas';
export { default as Checkbox } from './Checkbox';
export { default as Component } from './Component';
export { default as ContentColumn } from './ContentColumn';
export { default as ContentPane } from './ContentPane';
export { default as D3Path } from './D3Path';
export { default as Divider } from './Divider';
export { default as Dropzone } from './Dropzone';
export { default as FormErrors } from './FormErrors';
export { default as HintButton } from './HintButton';
export { default as HintControl } from './HintControl';
export { default as HintLabel } from './HintLabel';
export { default as Input } from './Input';
export { default as MenuButton } from './MenuButton';
export { default as ModalButton } from './ModalButton';
export { default as Nav } from './Nav';
export { default as Number } from './Number';
export { default as Select } from './Select';
export { default as Slider } from './Slider';
export { default as Table } from './Table';
export { default as ValueLabel } from './ValueLabel';
