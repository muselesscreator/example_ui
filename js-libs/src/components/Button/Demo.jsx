import React from 'react';
import { ButtonGroup } from 'semantic-ui-react';

import Component from '../Component';

import Button from './Button';

export class Demo extends Component {
  state = {
    toggleVal: false,
  }
  toggleVal = () => this.setState({ toggleVal: !this.state.toggleVal });
  render() {
    return (
      <div>
        <h4>Styling</h4>
        <Button>Simple Dummy Button</Button>
        <Button color="blue">Colored Button</Button>
        <Button color="blue" inverted={false}>Bold button</Button>
        <Button icon="cogs">Icon Button</Button>
        <Button icon="expand" />
        <Button active>Active button</Button>
        <h4>Button Group</h4>
        <ButtonGroup>
          <Button icon="expand" />
          <Button icon="map">Icon Button</Button>
          <Button color="green">Colored Button</Button>
        </ButtonGroup>
        <h4>Connected: Toggle value = {this.state.toggleVal ? 'True' : 'False'}</h4>
        <Button onClick={this.toggleVal}>Toggle Value</Button>
      </div>
    );
  }
}

export default Demo;
