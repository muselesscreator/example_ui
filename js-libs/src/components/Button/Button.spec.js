import React from 'react';
import { render, simpleTest } from 'modules/redux-test/components';

import { RCButton as Button } from './Button';

jest.mock('semantic-ui-react');

describe('Button component', () => {
  let el;
  const props = { some: 'component', props: 'for testing' };
  const className = 'cl_n';
  const icon = 'Icon';
  const children = <div>Some Stuff</div>

  simpleTest('basic', Button, { className, icon, ...props })
  simpleTest('with_children', Button, { className, icon, children, ...props })

  describe('className', () => {
    it('uses passed className alongside brw-btn className', () => {
      el = render(Button, { className });
      expect(el.hasClass('brw-btn')).toEqual(true);
      expect(el.hasClass(className)).toEqual(true);
    });
    const testClassName = (testClass) => {
      it(`includes ${testClass} className iff that prop is included`, () => {
        el = render(Button, { className });
        expect(el.hasClass(testClass)).toEqual(false);
        el = render(Button, { className, [testClass]: true });
        expect(el.hasClass(testClass)).toEqual(true);
      });
    }
    testClassName('active');
    testClassName('danger');
    testClassName('dark');
    testClassName('secondary');
  });

  describe('icon and children', () => {
    it('passes icon as a prop if no children are passed', () => {
      el = render(Button, { className, icon });
      expect(el.prop('icon')).toEqual(icon);
      expect(el.children().exists()).toEqual(false)
    });
    it('passes icon as a child if other children are passed', () => {
      el = render(Button, { className, icon, children });
      expect(el.prop('icon')).toEqual(undefined);
      expect(el.children().contains(children));
      const iconEl = el.children().find('Icon');
      expect(iconEl.exists()).toEqual(true);
      expect(iconEl.prop('name')).toEqual(icon);
    });
  });
});
