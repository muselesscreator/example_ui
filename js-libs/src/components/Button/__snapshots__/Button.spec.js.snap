// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component :basic 1`] = `
<Button
  as="button"
  className="cl_n brw-btn"
  icon="Icon"
  props="for testing"
  size="mini"
  some="component"
/>
`;

exports[`Button component :with_children 1`] = `
<Button
  as="button"
  className="cl_n brw-btn"
  props="for testing"
  size="mini"
  some="component"
>
  <Icon
    as="i"
    name="Icon"
  />
  <div>
    Some Stuff
  </div>
</Button>
`;
